app/services/provider_interface/setup_provider_relationship_permissions.rb

```diff
@@ -2,27 +2,17 @@ module ProviderInterface
   class PermissionsSetupError < StandardError; end
 
   class SetupProviderRelationshipPermissions
-    def self.call(permissions_data = {})
+    def self.call(models)
       ProviderRelationshipPermissions.transaction do
-        permissions_data.each do |id, permissions|
-          record = ProviderRelationshipPermissions.find(id)
+        models.map do |record|
           raise PermissionsSetupError, 'Permissions record has already been setup' if record.setup_at.present?
 
-          record.update!(permissions_attributes_for_persistence(permissions))
```

Comment:
Does `save!`  provider the same audit trail info about attributes? I'm gonna assume it does (the docs don't seem to show an example of this).

Reply:
Good question, checking this...

Reply:
Looks OK:

![Screenshot 2020-09-07 at 16 23 38](https://user-images.githubusercontent.com/642279/92401887-8b84aa00-f126-11ea-92a2-6923dbcfa1de.png)